engine/src/test/scala/cromwell/CromwellTestkitSpec.scala

```diff
@@ -258,6 +259,43 @@ abstract class CromwellTestkitSpec extends TestKit(new CromwellTestkitSpec.TestW
   implicit val defaultPatience = PatienceConfig(timeout = Span(30, Seconds), interval = Span(100, Millis))
   implicit val ec = system.dispatcher
 
+  // Allow to use shouldEqual between 2 WdlTypes while acknowledging for edge cases
+  implicit val wdlTypeSoftEquality = new Equality[WdlType] {
+    override def areEqual(a: WdlType, b: Any): Boolean = (a, b) match {
+      case (WdlStringType | WdlFileType, WdlFileType | WdlStringType) => true
+      case (arr1: WdlArrayType, arr2: WdlArrayType) => areEqual(arr1.memberType, arr2.memberType)
+      case (map1: WdlMapType, map2: WdlMapType) => areEqual(map1.valueType, map2.valueType)
+      case _ => a == b
+    }
+  }
+
+  // Allow to use shouldEqual between 2 WdlValues while acknowledging for edge cases and checking for WdlType compatibilty
+  implicit val wdlEquality = new Equality[WdlValue] {
+    def fileEquality(f1: String, f2: String) = Paths.get(f1).getFileName == Paths.get(f2).getFileName
+
+    override def areEqual(a: WdlValue, b: Any): Boolean = {
+      val typeEquality = b match {
+        case v: WdlValue => wdlTypeSoftEquality.areEqual(a.wdlType, v.wdlType)
+        case _ => false
+      }
+
+      val valueEquality = (a, b) match {
+        case (_: WdlFile, expectedFile: WdlFile) => fileEquality(a.valueString, expectedFile.valueString)
+        case (_: WdlString, expectedFile: WdlFile) => fileEquality(a.valueString, expectedFile.valueString)
+        case (array: WdlArray, expectedArray: WdlArray) => array.value.zip(expectedArray.value).map(Function.tupled(areEqual)).fold(true)(_ && _)
+        case (map: WdlMap, expectedMap: WdlMap) =>
+          val mapped = map.value.map {
```

Comment:
Similarly, `WdlMap`s may want to check that the keys contain the same size/elements. Right now `expectedMap` can actually be a superset of `map` and I think this method will return `true`.